insights/client/connection.py

```diff
@@ -616,92 +661,87 @@ def do_group(self):
     def _legacy_api_registration_check(self):
         '''
         Check registration status through API
+
+        Returns:
+            unreg_date if unregistered
+            True if registered
+            None is unregistered
```

Comment:
If I understand it correctly, _None_ means that the host [cannot be found](https://github.com/RedHatInsights/insights-core/blob/bb04fb85035fbb0eb8bab72486328bb2ad8cc9d1/insights/client/connection.py#L681) (404). A timestamp is returned if the host [has been found](https://github.com/RedHatInsights/insights-core/blob/bb04fb85035fbb0eb8bab72486328bb2ad8cc9d1/insights/client/connection.py#L697), but unregistered.

```suggestion
            unreg_date if unregistered
            True if registered
            None if not registered
```